src/gsy_e/models/strategy/__init__.py

```diff
@@ -909,7 +909,8 @@ def get_posted_bids(self, market: "Market", time_slot: Optional[DateTime] = None
 
     def _assert_market_type_on_bid_event(self, market_id):
         assert (ConstSettings.IAASettings.MARKET_TYPE == SpotMarketTypeEnum.TWO_SIDED.value or
-                self.area.is_market_future(market_id)), (
+                self.area.is_market_future(market_id) or
+                self.area.is_market_settlement(market_id)), (
             "Invalid state, cannot receive a bid if single sided market is globally configured or "
             "if it is not a future market bid.")
```

Comment:
I somehow knew that the commit message was not enough to explain this. By default, future and settlement markets are two-sided. They also have the ability to operate in parallel with one-sided spot market. In this case though, we will receive events from bid, delete bid etc. while running one-sided markets. This assertion used to validate that, when running an one-sided market, we do not receive any bid-related events, which is not true anymore as said above, since the future and settlement markets will send their own bids. Therefore we needed to change the assertion to accept bids from future and settlement markets even during one-sided spot market. 
Regarding the message per se, I think that the only thing that needs to change is the second line `if it is not a future or settlement market bid.` . Other than that looks clear to my eyes

Reply:
Thanks a lot for being patient and explain! For some reason my logic failed me earlier. I understand now what `_assert_market_type_on_bid_event` is handling and agree with your slight change of the assert message.